packages/browser-wallet/src/popup/pages/Setup/ChooseNetwork.tsx

```diff
@@ -3,33 +3,59 @@ import React from 'react';
 import { useNavigate } from 'react-router-dom';
 import Button from '@popup/shared/Button';
 import { absoluteRoutes } from '@popup/constants/routes';
-import { useSetAtom } from 'jotai';
+import { useSetAtom, useAtomValue } from 'jotai';
 import { useTranslation } from 'react-i18next';
+import { isRecoveringAtom } from '@popup/state';
 import { networkConfigurationAtom } from '@popup/store/settings';
-import { mainnet, testnet } from '../NetworkSettings/NetworkSettings';
+import { setupRoutes } from './routes';
+// TODO Remove stagenet
+import { mainnet, testnet, stagenet } from '../NetworkSettings/NetworkSettings';
 
 export function ChooseNetwork() {
     const navigate = useNavigate();
     const { t } = useTranslation('setup');
     const setNetworkConfiguration = useSetAtom(networkConfigurationAtom);
+    const isRecovering = useAtomValue(isRecoveringAtom);
+
+    const goToNext = () => {
+        if (isRecovering) {
+            navigate(`${absoluteRoutes.setup.path}/${setupRoutes.performRecovery}`);
+        } else {
+            navigate(absoluteRoutes.home.identities.path);
+        }
+    };
 
     return (
         <>
             <PageHeader>Choose a network</PageHeader>
             <div className="onboarding-setup__page-with-header">
                 <div className="onboarding-setup__page-with-header__description">
-                    <p>
-                        {t('chooseNetwork.descriptionP1')} <i>{t('chooseNetwork.descriptionP2')}</i>
-                    </p>
-                    <p>{t('chooseNetwork.descriptionP3')}</p>
+                    {!isRecovering && (
+                        <>
+                            <p>
+                                {t('chooseNetwork.create.descriptionP1')}{' '}
+                                <i>{t('chooseNetwork.create.descriptionP2')}</i>
+                            </p>
+                            <p>{t('chooseNetwork.create.descriptionP3')}</p>
+                        </>
+                    )}
+                    {isRecovering && (
+                        <>
+                            <p>
+                                {t('chooseNetwork.create.descriptionP1')}{' '}
```

Comment:
```suggestion
                                {t('chooseNetwork.restore.descriptionP1')}{' '}
```
The same for the other descriptions.

Reply:
:thinking: